modules/gapi/src/streaming/onevpl/cfg_param_device_selector.hpp

```diff
@@ -0,0 +1,41 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+//
+// Copyright (C) 2021 Intel Corporation
+
+#ifndef GAPI_STREAMING_ONEVPL_CFG_PARAM_DEVICE_SELECTOR_HPP
+#define GAPI_STREAMING_ONEVPL_CFG_PARAM_DEVICE_SELECTOR_HPP
+
+#include <opencv2/gapi/streaming/onevpl/device_selector_interface.hpp>
+#include <opencv2/gapi/streaming/onevpl/cfg_params.hpp>
+#include <opencv2/gapi/streaming/onevpl/source.hpp>
+
+#include "opencv2/gapi/own/exports.hpp" // GAPI_EXPORTS
+
+namespace cv {
+namespace gapi {
+namespace wip {
+namespace onevpl {
+
+struct GAPI_EXPORTS CfgParamDeviceSelector final: public IDeviceSelector {
+    CfgParamDeviceSelector(const CfgParams& params = {});
+    CfgParamDeviceSelector(Device::Ptr device_ptr,
+                           Context::Ptr ctx_ptr,
+                           const CfgParams& params);
+    ~CfgParamDeviceSelector();
+
+    DeviceScoreTable select_devices() const override;
+    DeviceScoreTable select_spare_devices() const override;
+    Context select_context(const DeviceScoreTable& selected_devices) override;
+    Context get_last_context() const override;
```

Comment:
Where these types are defined? Should they be `&`?

Reply:
declared/defined in device_selector_interface.hpp/cpp

 about `&` - In current context Context is a pointer & enum type

P.S. This  interface slightly modified in new PR in a way

```
    using DeviceScoreTable = std::map<Score, Device>;
    using DeviceContexts = std::vector<Context>;

    virtual ~IDeviceSelector();
    virtual DeviceScoreTable select_devices() const = 0;
    virtual DeviceContexts select_context() = 0;
```